### ISSUE TITLE
`alt!` parser somehow assumes `&[u8]` input

### ISSUE THREAD

**Issue by doy** [Geal/nom#101]
```
#[macro_use]
extern crate nom;
extern crate regex;

pub mod foo {
    use ::regex as regex;

    named!(re1<&str,&str>, re_find!(r"^foo"));
    named!(re2<&str,&str>, re_find!(r"^bar"));
    named!(either_re<&str,&str>, alt!(re1 | re2));

    pub fn regex_test() {
        let _ = either_re("foo");
    }
}
```

This throws an error when compiling:
```
<nom macros>:9:14: 9:15 error: mismatched types:
 expected `&[u8]`,
    found `&str`
(expected slice,
    found str) [E0308]
<nom macros>:9 $ submac ! ( i , $ ( $ args ) * ) } ) ; (
                            ^
<nom macros>:1:1: 40:74 note: in expansion of alt_parser!
<nom macros>:37:36: 37:56 note: expansion site
<nom macros>:1:1: 40:74 note: in expansion of alt_parser!
<nom macros>:31:1: 31:38 note: expansion site
<nom macros>:1:1: 40:74 note: in expansion of alt_parser!
<nom macros>:9:36: 9:73 note: expansion site
<nom macros>:1:1: 40:74 note: in expansion of alt_parser!
<nom macros>:2:1: 2:55 note: expansion site
<nom macros>:1:1: 40:74 note: in expansion of alt_parser!
<nom macros>:2:3: 2:40 note: expansion site
<nom macros>:1:1: 2:46 note: in expansion of alt!
<nom macros>:9:1: 9:34 note: expansion site
<nom macros>:1:1: 35:55 note: in expansion of named!
src/lib.rs:10:5: 10:51 note: expansion site
<nom macros>:9:14: 9:15 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to previous error
```

It's unclear what would be making it think there should be a `&[u8]` involved.

**Comment by Geal** [comment]
That one is tricky :)

The `&[u8]` does not come from `alt!` but from the definition of `Err` in `IResult::Error`. The error can contain position information, but it only supports `&[u8]`.

Fortunately, it is fixed in the 1.0 branch, with errors generic over the error code, and the position. I'll notify you once it is released.

**Comment by Geal** [comment]
Fixed in 1.0
